qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -1068,6 +1068,195 @@ def GaussianSquareDrag(
     return instance
 
 
+def GaussianSquareEcho(
+    duration: Union[int, ParameterExpression],
+    amp: Union[float, ParameterExpression],
+    sigma: Union[float, ParameterExpression],
+    width: Optional[Union[float, ParameterExpression]] = None,
+    angle: Optional[Union[float, ParameterExpression]] = 0.0,
+    active_amp: Optional[Union[float, ParameterExpression]] = 0.0,
+    active_angle: Optional[Union[float, ParameterExpression]] = 0.0,
+    risefall_sigma_ratio: Optional[Union[float, ParameterExpression]] = None,
+    name: Optional[str] = None,
+    limit_amplitude: Optional[bool] = None,
+) -> SymbolicPulse:
+    """An echoed Gaussian square pulse with an active tone overlaid on it.
+    
+    Exactly one of the ``risefall_sigma_ratio`` and ``width`` parameters has to be specified.
+    If ``risefall_sigma_ratio`` is not ``None`` and ``width`` is ``None``:
+    .. math::
+        \\text{risefall} &= \\text{risefall_sigma_ratio} \\times \\text{sigma}\\\\
+        \\text{width} &= \\text{duration} - 2 \\times \\text{risefall}
+    If ``width`` is not None and ``risefall_sigma_ratio`` is None:
+    .. math:: \\text{risefall} = \\frac{\\text{duration} - \\text{width}}{2}
+    The Gaussian Square Echo pulse is composed of three pulses. First, a Gaussian Square pulse
+    :math: `f_{echo}(x)` with amplitude ``amp`` and phase ``angle`` playing for half duration,
+    followed by a second Gaussian Square pulse :math: `-f_{echo}(x)` with opposite amplitude
+    and same phase playing for the rest of the duration. Third a Gaussian Square pulse
+     :math: `f_{active}(x)` with amplitude ``active_amp`` and phase ``active_angle``
+    playing for the entire duration. The Gaussian Square Echo pulse :math: `g_e()`
+    can be written as:
+    .. math::
+
+        g_e(x) &= \\begin{cases}\
+            f_{\\text{active}} + f_{\\text{echo}}(x)\
+                & x < \\frac{\\text{duration}}{2}\\\\
+            f_{\\text{active}} - f_{\\text{echo}}(x)\
+                & \\frac{\\text{duration}}{2} < x\\\\
+
+
+
+    References:
+        1. |citation1|_
+
+        .. _citation1: https://journals.aps.org/prxquantum/abstract/10.1103/PRXQuantum.1.020318
+
+        .. |citation1| replace:: *Sundaresan, N., Lauer, I., Pritchett, E.,
+           Magesan, E.,  Jurcevic, P. & Gambetta, J. M.
+           Reducing Unitary and Spectator Errors in Cross Resonance with
+           Optimized Rotary Echoes. PRX Quantum 1, 020318 (2020).*
+    Args:
+        duration: Pulse length in terms of the sampling period `dt`.
+        amp: The amplitude of the rise and fall and of the square pulse.
+        sigma: A measure of how wide or narrow the risefall is; see the class
+               docstring for more details.
+        width: The duration of the embedded square pulse.
+        angle: The angle in radians of the complex phase factor uniformly
+            scaling the pulse. Default value 0.
```

Comment:
I have tried to be more specific in the docstring on what refers to what. I'd like to avoid changing variable names as `amp` and `angle` are the standard variable names used for other pulses (and one could define a GaussianSquareEcho just by defining these two).